include/albatross/src/core/distribution.hpp

```diff
@@ -100,7 +100,28 @@ struct MarginalDistribution : public DistributionBase<MarginalDistribution> {
   }
 
   bool operator==(const MarginalDistribution &other) const {
-    return (mean == other.mean && covariance == other.covariance);
+    const double epsilon = 0.1;
```

Comment:
Had to set it to be fairly loose to support https://github.com/swift-nav/orion-engine/pull/6673 

Any concerns about it? I could just write a custom comparator that lets people pass in `epsilon` if there are

Reply:
I fear people might assume that `x == y` would provide a strict equality, but I do see the value in enabling inexact equality checks. What about having the syntax be something like:
```
x.approx_equal(y, 1e-6)
```
?

Reply:
Done